plugins/ExtendedTools/objmgr.c

```diff
@@ -1158,64 +1300,65 @@ NTSTATUS EtEnumCurrentDirectoryObjects(
 
     PRESOLVER_THREAD_CONTEXT threadContext = PhAllocateZero(sizeof(RESOLVER_THREAD_CONTEXT));
     threadContext->Context = Context;
-    Context->BreakResolverThread = &threadContext->Break;
+    Context->BreakResolverThread = (PBOOLEAN)&threadContext->Break;
 
     PhCreateThread2(EtpTargetResolverThreadStart, threadContext);
 
     return STATUS_SUCCESS;
 }
 
 VOID EtObjectManagerFreeListViewItems(
-    _In_ POBJECT_CONTEXT Context
+    _In_ PET_OBJECT_CONTEXT Context
     )
 {
     INT index = INT_ERROR;
 
     if (Context->BreakResolverThread)
     {
-        *Context->BreakResolverThread = TRUE;
+        WriteRelease8(Context->BreakResolverThread, TRUE);
```

Comment:
🤔 - this reads dangerous to me, I think I can reason this away by saying a new thread will not be created and race here? I suppose this is fine then... my initial read is that we could race here and set the wrong boolean context. Or worse a double-fetch of the value resulting in a null pointer de-reference.

Reply:
This code executes from UI thread only. For example, user press refresh button. Let's assume what resolver still working.

```
>	ExtendedTools.dll!EtObjectManagerFreeListViewItems(_ET_OBJECT_CONTEXT * Context) Строка 1317	C
 	ExtendedTools.dll!EtpObjectManagerRefresh(_ET_OBJECT_CONTEXT * Context) Строка 2367	C
 	ExtendedTools.dll!WinObjDlgProc(HWND__ * hwndDlg, unsigned int uMsg, unsigned __int64 wParam, __int64 lParam) Строка 2830	C
```
We send break command to current resolver thread, refresh tree and reselect previously active directory. After `EtpObjectManagerRefresh` calls `EtEnumCurrentDirectoryObjects`, which creates new resolver thread and store its `Break` switch pointer in BreakResolverThread.

```
Context->BreakResolverThread = (PBOOLEAN)&threadContext->Break;

PhCreateThread2(EtpTargetResolverThreadStart, threadContext);
```
Yes, it could possible be a race. Thanks for nit. I moved clearing of `context->BreakResolverThread` inside if block, so the pointer only be cleared if resolver wasn't imterrupted.
```
    if (status != STATUS_ABANDONED && !ReadAcquire8(&threadContext->Break))
    {
		...
        context->BreakResolverThread = NULL;
    }
```